src/core/server/status/routes/status.ts

```diff
@@ -0,0 +1,189 @@
+      memory: {
+        heap: {
+          total_in_bytes: number;
+          used_in_bytes: number;
+          size_limit: number;
+        };
+        resident_set_size_in_bytes: number;
+      };
+      event_loop_delay: number;
+      pid: number;
+      uptime_in_millis: number;
+    };
+    os: {
+      load: Record<string, number>;
+      memory: {
+        total_in_bytes: number;
+        used_in_bytes: number;
+        free_in_bytes: number;
+      };
+      uptime_in_millis: number;
+      platform: string;
+      platformRelease: string;
+    };
+    response_times: {
+      max_in_millis: number;
+    };
+    requests: {
+      total: number;
+      disconnects: number;
+      statusCodes: Record<number, number>;
+      status_codes: Record<number, number>;
+    };
+    concurrent_connections: number;
+  };
+}
+
+const serializeStatus = (status: ServiceStatus): SerializableServiceStatus => ({
+  ...status,
+  level: status.level.toString(),
+});
+
+const serializeStatusRecord = (
+  statuses: Record<string, ServiceStatus>
+): Record<string, SerializableServiceStatus> =>
+  Object.keys(statuses).reduce((acc, serviceName) => {
+    acc[serviceName] = serializeStatus(statuses[serviceName]);
+    return acc;
+  }, {} as Record<string, SerializableServiceStatus>);
+
+export const registerStatusRoute = ({ router, config, metrics, status }: Deps) => {
+  router.get(
+    {
+      path: '/api/status',
+      options: {
+        authRequired: !config.allowAnonymous,
+      },
+      validate: {
+        query: schema.object({
+          v8format: schema.boolean({ defaultValue: false }),
+        }),
+      },
+    },
+    async (context, req, res) => {
+      const { version, buildSha, buildNum } = config.packageInfo;
+      const versionWithoutSnapshot = version.replace(SNAPSHOT_POSTFIX, '');
+
+      let statusInfo: LegacyStatusInfo | StatusInfo;
+      if (req.query?.v8format) {
+        const [overall, core, plugins] = await combineLatest([
+          status.overall$,
+          status.core$,
+          status.plugins$,
+        ])
+          .pipe(first())
+          .toPromise();
+        statusInfo = {
+          overall: serializeStatus(overall),
+          core: serializeStatusRecord((core as unknown) as Record<string, ServiceStatus>),
+          plugins: serializeStatusRecord(plugins),
+        };
+      } else {
+        statusInfo = await calculateLegacyStatus({
+          ...status,
+          versionWithoutSnapshot,
+        });
+      }
+
+      const lastMetrics = await metrics.getOpsMetrics$().pipe(first()).toPromise();
+
+      const body: StatusHttpBody = {
+        name: config.serverName,
+        uuid: config.uuid,
+        version: {
+          number: versionWithoutSnapshot,
+          build_hash: buildSha,
+          build_number: buildNum,
+          build_snapshot: SNAPSHOT_POSTFIX.test(version),
+        },
+        status: statusInfo,
+        metrics: {
+          last_updated: new Date().toISOString(),
```

Comment:
this is not entirely true. `last_updated` can differ from the actual metric collection in `metrics.collectionInterval` (5s by default).

Reply:
To solve this and the other comment, I may do a `metrics.getOpsMetrics$().subscribe()` call outside the route handler and keep track of the last updated value. This is similar to how the `/api/stats` endpoint was already working.